codecov_auth/authentication/repo_auth.py

```
@@ -253,86 +256,64 @@ def authenticate_credentials(self, token):
         service = match.group(1)
         try:
             service_enum = Service(service)
-            # Currently only Github is supported
-            # TODO [codecov/engineering-team#914]: Extend tokenless support to other providers
-            if service_enum != Service.GITHUB:
-                raise exceptions.AuthenticationFailed(self.auth_failed_message)
         except ValueError:
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
+            return None, None
+
         # Validate that next group exists and decode slug
         encoded_slug = match.group(2)
         repo = get_repository_from_string(service_enum, encoded_slug)
+
         if repo is None:
             # Purposefully using the generic message so that we don't tell that
             # we don't have a certain repo
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
-        return repo
+            return None, None
 
-    @async_to_sync
-    async def get_pull_request_info(self, repository_service, fork_pr: str):
-        try:
-            return await repository_service.get_pull_request(fork_pr)
-        except TorngitObjectNotFoundError:
-            raise exceptions.AuthenticationFailed(self.auth_failed_message)
-        except TorngitRateLimitError as e:
-            metrics.incr("auth.get_pr_info.rate_limit_hit")
-            sentry_metrics.incr("auth.get_pr_info.rate_limit_hit")
-            if e.reset:
-                now_timestamp = datetime.now().timestamp()
-                retry_after = int(e.reset) - int(now_timestamp)
-            elif e.retry_after:
-                retry_after = int(e.retry_after)
-            raise exceptions.Throttled(
-                wait=retry_after, detail=self.rate_limit_failed_message
-            )
+        commitid = match.group(3)
+
+        return repo, commitid
+
+    def get_branch(self, request, commitid=None):
+        if commitid:
+            commit = Commit.objects.get(commitid=commitid)
```

I think `.get` will throw an error if this doesn't exist instead of `None`